src/core/radio/radio_platform.cpp

```diff
@@ -212,6 +212,11 @@ OT_TOOL_WEAK void otPlatRadioSetMacFrameCounter(otInstance *aInstance, uint32_t
     OT_UNUSED_VARIABLE(aMacFrameCounter);
 }
 
+OT_TOOL_WEAK void otPlatRadioSetMacFrameCounterIfLarger(otInstance *aInstance, uint32_t aMacFrameCounter)
+{
+    otPlatRadioSetMacFrameCounter(aInstance, aMacFrameCounter);
```

Comment:
Maybe add a warning comment here that this weak implementation breaks intended functionality for the corner case?